doc/test_organization/managing_tests_dependencies.qbk

```diff
@@ -7,17 +7,31 @@
 
 [section:tests_dependencies Managing test dependencies]
 
-Dependencies among tests are managed by means of [link boost_test.tests_organization.decorators decorators]. 
+In general, you should make sure that test cases are independent on one another.
+For any two test cases `TA` and `TB`, `TB` must not take for granted that `TA` has already executed,
+even if `TA` is declared before `TB` in the same translation unit.
+
+The only ordering-related guarantee that __UTF__ makes by default is that if test cases `TA` and `TB`
+are declared in the same test suite, no test case (call it `TX`) from any other test suite is executed between `TA` and `TB`,
+even if the declaration of `TX` appears between the declarations of `TA` and `TB`.
+In other words, all tests from a suite are executed in one go, even if the test suite namespace is opened multiple times.
```

Comment:
2015-05-22 23:53 GMT+02:00 Raffi Enficiaud <notifications@github.com>:

> In doc/test_organization/managing_tests_dependencies.qbk
> <https://github.com/boostorg/test/pull/64#discussion_r30933561>:
>
> > @@ -7,17 +7,31 @@
> >
> >  [section:tests_dependencies Managing test dependencies]
> >
> > -Dependencies among tests are managed by means of [link boost_test.tests_organization.decorators decorators].
> > +In general, you should make sure that test cases are independent on one another.
> > +For any two test cases `TA` and `TB`, `TB` must not take for granted that `TA` has already executed,
> > +even if `TA` is declared before `TB` in the same translation unit.
> > +
> > +The only ordering-related guarantee that __UTF__ makes by default is that if test cases `TA` and `TB`
> > +are declared in the same test suite, no test case (call it `TX`) from any other test suite is executed between `TA` and `TB`,
> > +even if the declaration of `TX` appears between the declarations of `TA` and `TB`.
> > +In other words, all tests from a suite are executed in one go, even if the test suite namespace is opened multiple times.
>
> Is this really a guaranty? personnally I do not know (so I'm not
> questioning your sentence but the framework)
>

I think we need to guarantee that in order for the "suite entry/exit
fixture" to make sense. This is what the entry/exit fixture guarantees:

first suite setup, then all suite tests, last suite tear down. So, this
precludes any shuffling of tests between suites (but allows shufflig within
one suite).